CharlieBackend.Core/Models/ResultModel/ErrorCode.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Core.Models.ResultModel
+{
+    public enum ErrorCode
+    {
+        None = 0,
+        NullReference,
+        InternalError,
+        DataNotFound,
+        Unauthorised,
+        NotEnoughtRights,
+        BadRequest,
+        Conflict,
+        Forbidden,
+        UnprocessableEntity,
+        ValidationProblem,
+        ManualError,
```

Comment:
This codes should not represent direct mapping to HTTP status codes, neither some internal reasons such as NullReferenceException. Please leave only generic ValidationError and Unauthorised for now, we'll add more specific codes after adding new validations in services.